app.rb

```diff
@@ -79,23 +78,33 @@
   @page_title = @term[:name]
 
   last_sha = @house[:sha]
-  csv_file = EveryPolitician::GithubFile.new(@term[:csv], last_sha)
-  @csv = CSV.parse(csv_file.raw, headers: true, header_converters: :symbol, converters: nil)
-
-  person_ids = @csv.map { |row| row[:id] }.uniq
 
   popolo_file = EveryPolitician::GithubFile.new(@house[:popolo], last_sha)
   popolo = EveryPolitician::Popolo.parse(popolo_file.raw)
-  people = popolo.persons.find_all { |p| person_ids.include?(p.id) }.sort_by { |p| p.sort_name }
 
-  @parties = @csv.find_all { |r| r[:end_date].to_s.empty? || r[:end_date] == @term[:end_date] }
-                 .group_by { |r| r[:group] }
-                 .sort_by { |p, ms| [-ms.count, p] }
+  memberships = popolo.memberships.find_all { |m| m.legislative_period_id.split('/').last == id }
+
+  person_ids = memberships.map(&:person_id)
+  people = popolo.persons.find_all { |p| person_ids.include?(p.id) }.sort_by(&:sort_name)
 
-  memberships_by_person = popolo.memberships.find_all { |m| m.legislative_period_id == "term/#{id}" }.group_by(&:person_id)
+  memberships_by_person = memberships.group_by(&:person_id)
   areas_by_id = Hash[popolo.areas.map { |a| [a.id, a] }]
   orgs_by_id = Hash[popolo.organizations.map { |o| [o.id, o] }]
 
+  current_memberships = memberships.find_all { |mem| mem.end_date.to_s.empty? || mem.end_date == @term[:end_date] }
+                                   .group_by(&:on_behalf_of_id)
+                                   .map { |group_id, mems| [orgs_by_id[group_id], mems] }
+                                   .sort_by { |group, mems| [-mems.count, group.name] }
+
```

Comment:
`current_memberships` seems remarkably complex when all we're using it for now is creating `@parties`